packages/web/components/templates/article/HighlightsLayer.tsx

```diff
@@ -454,6 +455,30 @@ export function HighlightsLayer(props: HighlightsLayerProps): JSX.Element {
             })
           }
           break
+        case 'copy': {
+          const selection = window.getSelection()
+          if (selection === null) return
+
+          const userSelectionText = selection.toString()
+          let textToCopy = ''
+
+          if (focusedHighlight) {
+            const highlightedElements = getHighlightElements(
+              focusedHighlight.id
+            )
+            highlightedElements.forEach(
+              (element) => (textToCopy += element.textContent)
+            )
+          } else if (userSelectionText) {
+            textToCopy = userSelectionText
+          }
+
+          if (textToCopy) await navigator.clipboard.writeText(textToCopy)
```

Comment:
We might want to add a success toast

Reply:
I guess we will want to use the `showSuccessToast` from `toastHelpers.tsx` for this? Is "Copied Successfully!" a good message? Might be a corner case (if user has the clipboard access disabled) but do we want to show an error toast as well?

Reply:
Yeah we can do an error toast too, something like:

```
if (success) {
   showSuccessToast('Highlight copied', {
            position: 'bottom-right',
          })
   } else {
   showErrorToast('Error copying highlight, permission denied.',  {
            position: 'bottom-right',
          })
```
   }

Reply:
Added the toasts :)